metadata-ingestion/src/datahub/ingestion/source/metadata/lineage.py

```diff
@@ -0,0 +1,179 @@
+import logging
+from dataclasses import dataclass, field
+from typing import Iterable, List, Optional, Union
+
+from pydantic import validator
+
+import datahub.metadata.schema_classes as models
+from datahub.cli.cli_utils import get_aspects_for_entity
+from datahub.configuration.common import ConfigModel
+from datahub.configuration.config_loader import load_config_file
+from datahub.emitter.mce_builder import (
+    get_sys_time,
+    make_dataset_urn_with_platform_instance,
+)
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit, UsageStatsWorkUnit
+
+logger = logging.getLogger(__name__)
+
+auditStamp = models.AuditStampClass(
+    time=get_sys_time(), actor="urn:li:corpUser:pythonEmitter"
+)
+
+
+class EntityConfig(ConfigModel):
+    name: str
+    type: str
+    env: str
+    platform: str
+    platform_instance: Optional[str]
+
+
+class EntityNodeConfig(ConfigModel):
+    entity: EntityConfig
+    upstream: Optional[List["EntityNodeConfig"]]
+
+
+# https://pydantic-docs.helpmanual.io/usage/postponed_annotations/ required for when you reference a model within itself
+EntityNodeConfig.update_forward_refs()
+
+
+class LineageFileSourceConfig(ConfigModel):
+    file: str
+    preserve_upstream: bool = True
+
+
+class DefaultConfig(ConfigModel):
```

Comment:
s/DefaultConfig/VersionedConfig/ ?

Reply:
I tried to remain consistent with a similar source https://github.com/tucows/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py#L51, but I'm not opposed to the new name. Let me know if you prefer the consistency or the proposed name

Reply:
Let's use more insightful names. We will fix the other one when we get a chance to touch it. In fact, this is a highly reusable config that it should be moved to https://github.com/linkedin/datahub/blob/master/metadata-ingestion/src/datahub/configuration/common.py.

Reply:
Sure, how does it sound to place 

```python
class VersionedConfig(ConfigModel):
    version: str = "1"
```
Within `common.py` and to leave validation to classes that use `VersionedConfig`?

Reply:
Yes. Configs that inherit from this can perform their own validation and changing the default value as needed.